docs/NON_DEVCONTAINER_SETUP.md

```diff
@@ -1,54 +1,80 @@
+  - [Pylance](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance)
+  - [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+  - [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDevApp.ms-teams-vscode-extension) **Optional**
+- [Python 3.11](https://www.python.org/downloads/release/python-3119/)
+- [Node.js LTS](https://nodejs.org/en)
+- [Azure Developer CLI](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/install-azd)
+- [Azure Functions Core Tools](https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local)
 
+## Setup
 
-### Running the sample using the Azure Developer CLI (azd)
+1. Configure your Python environment:
 
-The Azure Developer CLI (`azd`) is a developer-centric command-line interface (CLI) tool for creating Azure applications.
+    ```bash
+    pip install --upgrade pip
+
+    pip install poetry
+
+    # https://pypi.org/project/poetry-plugin-export/
+    pip install poetry-plugin-export
+
+    poetry env use python3.11
+
+    poetry config warnings.export false
+
+    poetry install --with dev
+    ```
+
+1. Install `pre-commit` Git hooks:
 
-You need to install it before running and deploying with the Azure Developer CLI. (If you use the devcontainer, everything is already installed)
+    ```bash
+    poetry run pre-commit install
+    ```
 
-### Windows
+1. Install Node.js dependencies:
 
-```powershell
-powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"
-```
+    ```bash
+    (cd ./code/frontend; npm install)
+
+    (cd ./tests/integration/ui; npm install)
+    ```
+
+1. Select the Python interpreter in Visual Studio Code:
+
+    - Open the command palette (`Ctrl+Shift+P` or `Cmd+Shift+P`).
+    - Type `Python: Select Interpreter`.
+    - Select the Python 3.11 environment created by Poetry.
+
+### Running the sample using the Azure Developer CLI (azd)
+
+The Azure Developer CLI (`azd`) is a developer-centric command-line interface (CLI) tool for creating Azure applications.
 
-### Linux/MacOS
+1. Log in to Azure using `azd`:
 
-```
-curl -fsSL https://aka.ms/install-azd.sh | bash
-```
+  ```
+  azd auth login
+  ```
 
-After logging in with the following command, you will be able to use the `azd` cli to quickly provision and deploy the application.
+1. Execute the `azd init` command to initialize the environment and enter the solution accelerator name when prompted:
 
-```
-azd auth login
-```
+  ```
+  azd init -t chat-with-your-data-solution-accelerator
+  ```
 
-Then, execute the `azd init` command to initialize the environment (You do not need to run this command if you already have the code or have opened this in a Codespace or DevContainer).
-```
-azd init -t chat-with-your-data-solution-accelerator
-```
-Enter an environment name.
+1. Run `azd up` to provision all the resources to Azure and deploy the code to those resources.
```

Comment:
This numbered don't seem to be incrementing as expected. Do we need to remove the line break above?

Reply:
good spot, fixed